src/pybind/mgr/cephadm/ssh.py

```diff
@@ -135,16 +135,27 @@ async def _execute_command(self,
         cmd = "sudo " + " ".join(quote(x) for x in cmd)
         logger.debug(f'Running command: {cmd}')
         try:
-            r = await conn.run(cmd, input=stdin.decode() if stdin else None)
+            # The default value passed to encoding in asyncssh is an empty
+            # tuple. If an empty tuple is found when a channel is created, the
+            # encoding 'utf-8' is set automatically. To prevent that for
+            # `type(input) is bytes` we need to set it explicitly to None, so
+            # that no conversion whatsoever is done.
+            encoding = ()
+            if type(stdin) is bytes:
+                encoding = None
+            r = await conn.run(cmd, input=stdin if stdin else None, encoding=encoding)
         # handle these Exceptions otherwise you might get a weird error like TypeError: __init__() missing 1 required positional argument: 'reason' (due to the asyncssh error interacting with raise_if_exception)
         except (asyncssh.ChannelOpenError, Exception) as e:
             # SSH connection closed or broken, will create new connection next call
             logger.debug(f'Connection to {host} failed. {str(e)}')
             await self._reset_con(host)
             self.mgr.offline_hosts.add(host)
             raise OrchestratorError(f'Unable to reach remote host {host}. {str(e)}')
-        out = r.stdout.rstrip('\n')
-        err = r.stderr.rstrip('\n')
+        out = r.stdout
+        err = r.stderr
+        if type(r.stdout) is str:
```

Comment:
nit, i'd prefer check `encoding` instead of dispatching by the type of an object.

Reply:
.. like above simply assume raw bytes?